src/types/datetime.rs

```diff
@@ -356,6 +332,38 @@ pyobject_native_type!(
     PyTZInfo_Check
 );
 
+/// Equivalent to `datetime.timezone.utc`
+#[cfg(all(Py_3_7, not(PyPy)))]
+pub fn timezone_utc(py: Python) -> &PyTzInfo {
+    unsafe {
+        &*(&*ffi::PyDateTime_TimeZone_UTC as *const *mut ffi::PyObject
+            as *const crate::Py<PyTzInfo>)
+    }
+    .as_ref(py)
+}
+
+/// Equivalent to `datetime.timezone(offset)`
+#[cfg(all(Py_3_7, not(PyPy)))]
+pub fn timezone_from_offset<'py>(py: Python<'py>, offset: &PyDelta) -> PyResult<&'py PyTzInfo> {
+    unsafe { py.from_owned_ptr_or_err(ffi::PyTimeZone_FromOffset(offset.as_ptr())) }
+}
+
+/// Equivalent to `datetime.timezone(offset, name)`
+#[cfg(all(Py_3_7, not(PyPy)))]
+pub fn timezone_from_offset_and_name<'py>(
+    py: Python<'py>,
+    offset: &PyDelta,
+    name: &str,
+) -> PyResult<&'py PyTzInfo> {
+    let name = name.into_py(py);
+    unsafe {
+        py.from_owned_ptr_or_err(ffi::PyTimeZone_FromOffsetAndName(
+            offset.as_ptr(),
+            name.as_ptr(),
+        ))
+    }
+}
```

Comment:
Wouldn't these be easier to discover if we put it within `impl PyTzInfo`?

Reply:
I went back and forth on that. Yes, it probably would. However I wasn't sure if that was totally "correct" because they're actually creating a subtype. If you also think it'd be nicer to have them on `impl PyTzInfo` then I'm persuaded that's the right thing to do 👍

Reply:
No, I am just thinking that `datetime` module could have tons of functions already and separating these would increase those later on, since these will only create `PyTzInfo` I don't see why not put it there.

Reply:
I definitely don't think this should go in `impl PyTzInfo`. `PyTzInfo` exists only to be a base class, so you shouldn't really be able to get instances of it. Despite Rust's lack of nominal type hierarchies, it needs Rust wrappers so you can do stuff like this: https://github.com/PyO3/pyo3/blob/3ee84a3c26a67686d618cfec8c85463be7506bb0/examples/pyo3-pytests/src/datetime.rs#L181-L202

Reply:
I don't mean with `&self`, I just mean something like
```rust
impl PyTzinfo {
    fn utcoffset<'py>(py: Python<'py>, offset: &PyDelta) -> PyResult<&'py PyTzInfo> {
        ...
    }
}
```

Reply:
@pickfire Regardless, I don't think it's the right abstraction, because it implies that this is a *constructor* for `PyTzInfo`, which is an abstract base class.

I do think it would make sense for these methods to turn into a wrapper class for `datetime.timezone` like the others (minus the `_Check` methods and such), in which case these could be alternate constructors for `PyFixedTimezone` or whatever it's called.